### ISSUE TITLE
SPIKE: Determine if any refinements can be made on "loading" action in the event there are no holdings with an item

### ISSUE THREAD

**Issue by lovinscari** [emory-libraries/blacklight-catalog#1213]
We have found that when a record has no holdings, the real time availability "loading" action continues to run indefinitely for all search results. We believe cataloging has removed a number of the records with no holdings, but would also like application developers to determine if anything can be done to allow availability to show for all records with holding and perhaps deliver a different message (such as the contact us badge) in the event a record is found with no holdings. Please document the work needed within this ticket and any estimate of time (if known) so this work can be scheduled if the problem continues to exist after the Alma cleanup.

**Comment by rotated8** [comment]
For search results, we are requesting the availability data for all the items in the results at once. This allows us to make one API call instead of 10, 20, 50, or 100 (the default 'per page' options) at a time. If any item in the results does not have availability data (because it does not have holdings, or some other issue), then Alma returns an error and no availability information for any item.

Generally, I am against making queries for each record individually, since we have a cap of API requests we can make in a day. As we intend to move requesting from the Primo VE Services page into Blacklight via API, I worry that increasing our API usage outside of requesting will have adverse impacts.

Specifically, while we have default options of 10, 20, 50, and 100 results per page, that number is embedded in the search url and can be edited by the user. See [this link](https://search.libraries.emory.edu/?utf8=%E2%9C%93&per_page=17&search_field=keyword&q=pizza) with 17 results per page. Even if we got rid of the 100 results per page option, users could edit the url to get it back. Luckily, Blacklight caps the number of results it will accommodate at 100 (trying for 105, (this link)[https://search.libraries.emory.edu/?utf8=%E2%9C%93&per_page=105&search_field=keyword&q=pizza] is capped at 100). If we switched to individual availability requests, one user could easily spend 100 of our API requests even if we removed that option from the UI. A bad actor or a search bot could overload our system and not only decimate our API request allowance, but also tie the system up for other users as it tried to handle all the requests.

**Comment by abelemlih** [comment]
@lovinscari @rotated8 I agree with Collin's feedback, is there anything else you would like me to look into for this ticket, or is it ready to move to "To Accept"?

**Comment by abelemlih** [comment]
@lovinscari I am moving this ticket to "To Accept" based on Collin's feedback above. Let me know if there is anything else you would like us to look into regarding optimizing loading availability on the calendar. Thanks!

**Comment by lovinscari** [comment]
Thank you all for your work on this. I will update documentation for users but this will not move forward with development.
